osbenchmark/workload/loader.py

```diff
@@ -982,9 +991,24 @@ def read(self, workload_name, workload_spec_file, mapping_dir):
                 f.write(rendered)
             self.logger.info("Final rendered workload for '%s' has been written to '%s'.", workload_spec_file, tmp.name)
             workload_spec = json.loads(rendered)
+
         except jinja2.exceptions.TemplateNotFound:
             self.logger.exception("Could not load [%s]", workload_spec_file)
             raise exceptions.SystemSetupError("Workload {} does not exist".format(workload_name))
+
+        except jinja2.exceptions.TemplateSyntaxError as e:
```

Comment:
This is a really good improvement.  The implementation is fine for now, but moving the processing to a separate function with a map of error string patterns to suggestions would be helpful.  This could be in a separate module and can be updated if the Jinja2 parser changes.